### PR TITLE
Add MAGeCK tool for CRISPR

### PR DESCRIPTION
**Pull request by mblue9**
This PR is to add MAGeCK (https://sourceforge.net/p/mageck/wiki/Home/) for genome-wide CRISPR analysis to IUC.

I need to review this again to see if it's ok and if I should add some more tests, then will submit, hopefully next week.

Thanks to @bgruening for help with some fixes!

### CODE REVIEW

**Comment by bgruening** [comment]
Seems that the pdf could not be generated?
```
Error in texi2dvi("output_countsummary.tex", pdf = TRUE) :
pdflatex is not available
```

**Comment by mblue9** [comment]
Have taken this out of WIP as it's ready for review.

File: tools/mageck/mageck_count.xml [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+    </expand>
+    <expand macro="version" />
+    <command detect_errors="exit_code"><![CDATA[
+
+mageck count
+
+-l '$sgrna_library_file'
+
+#if $reads.format_select == "files":
+    --fastq ${ ' '.join( ['%s' % $x.sample for x in $reads.rep_sample] ) }
```
Comment (bgruening) [inline tools/mageck/mageck_count.xml]: can you use this instead: `${ ' '.join( ["'%s'" % $x.sample for $x in $reads.rep_sample] ) }`

File: tools/mageck/mageck_pathway.xml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        <param name="gene_ranking" argument="--gene-ranking" type="data" format="tabular" label="Gene Ranking file" help="The gene ranking file generated by the gene test step. Only one enrichment comparison will be performed." />
+        <param name="gmt_file" argument="--gmt-file" type="data" format="tabular" label="Pathway GMT file" help="The pathway file in GMT format. See Help below for more information" />
+
+        <section name="adv" title="Advanced Options">
+            <param name="single_ranking" argument="--single-ranking" type="boolean" truevalue="--single-ranking" falsevalue="" checked="false" optional="true"
+                label="Single ranking file"
+                help="The provided file is a (single) gene ranking file, either positive or negative selection. Only one enrichment comparison will be performed. Default: No" />
+            <param name="method" argument="--method" type="select" label="Method for testing pathway enrichment" >
+                <option value="gsea" selected="True">GSEA</option>
+                <option value="rra">RRA</option>
```
Comment (bgruening) [inline tools/mageck/mageck_pathway.xml]: maybe mention the meaning of RRA

File: tools/mageck/mageck_count.xml (hunk 1 above)
Comment (mblue9) [reply tools/mageck/mageck_count.xml]: ok have changed this and fixed the inputs to allow input of .gz and .bam (needs .gz, .bam file extensions) etc

File: tools/mageck/mageck_pathway.xml (hunk 2 above)
Comment (mblue9) [reply tools/mageck/mageck_pathway.xml]: done

**Comment by bgruening** [comment]
спасибо!

**Pull request merged**